### PR TITLE
Add release checklist

### PR DESCRIPTION
**Pull request by atravitz**
## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

**Review by IAlibay — APPROVED** [review]

**Review by IAlibay — CHANGES_REQUESTED** [review]
What about adding the conda lock and single file installers to the release assets?

**Comment by mikemhenry** [comment]
That step happens after the conda-forge build is done/uploaded. Adding the assets is automated now 😎 but still requires a manual click to fire off (for now, we could automate it as well). So something like "After the conda-forge package is uploaded, start single file installer and docker image workflow (and verify it was successful)"

File: .github/PULL_REQUEST_TEMPLATE/release_template.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+Make the PR:
+* [ ] Create a new release-prep branch corresponding to the version name, e.g. `release-prep-v1.2.0`.  Note: please follow [semantic versioning](https://semver.org/).
+* [ ] Check that all user-relevant updates are included in the `news/` rever `.rst` files. You can backfill any additional items by making a new .rst, e.g. `backfill.rst`
+* [ ] Run [rever](https://regro.github.io/rever-docs/index.html#), e.g. `rever 1.2.0`. This will auto-commit `docs/CHANGELOG.md` and remove the `.rst` files from `news/`. 
+* [ ] Verify that`docs/CHANGELOG.md` looks correct.
+* [ ] Make the PR and verify that CI/CD passes. 
+* [ ] Merge the PR into `main`.
+
+After Merging the PR [follow this guide]https://github.com/OpenFreeEnergy/openfe/releases/new
```
Comment (IAlibay) [inline .github/PULL_REQUEST_TEMPLATE/release_template.md:13]:
```suggestion
After Merging the PR [follow this guide](https://github.com/OpenFreeEnergy/openfe/wiki/How-to-create-a-new-release)
```
This?

**Review by IAlibay — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE/release_template.md (hunk 1 above)
Comment (atravitz) [reply .github/PULL_REQUEST_TEMPLATE/release_template.md]: 🤦yes.

**Review by IAlibay — APPROVED** [review]
lgtm!

**Pull request merged**